lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn2/src/main.c

```diff
@@ -0,0 +1,104 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/dapxsumkbn2.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/strided/base/stride2offset.h"
+#include <math.h>
+
+/**
+* Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using a second-order iterative Kahan–Babuška algorithm.
+*
+* ## Method
+*
+* -   This implementation uses a second-order iterative Kahan–Babuška algorithm, as described by Klein (2005).
+*
+* ## References
+*
+* -   Klein, Andreas. 2005. "A Generalized Kahan-Babuška-Summation-Algorithm." _Computing_ 76 (3): 279–93. doi:[10.1007/s00607-005-0139-x](https://doi.org/10.1007/s00607-005-0139-x).
+*
+* @param N        number of indexed elements
+* @param alpha    scalar constant
+* @param X        input array
+* @param strideX  stride length
+* @return         output value
+*/
+double API_SUFFIX(stdlib_strided_dapxsumkbn2)( const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX ) {
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX(stdlib_strided_dapxsumkbn2_ndarray)( N, alpha, X, strideX, ox );
+}
+
+/**
+* Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using a second-order iterative Kahan–Babuška algorithm and alternative indexing semantics.
+*
+* ## Method
+*
+* -   This implementation uses a second-order iterative Kahan–Babuška algorithm, as described by Klein (2005).
+*
+* ## References
+*
+* -   Klein, Andreas. 2005. "A Generalized Kahan-Babuška-Summation-Algorithm." _Computing_ 76 (3): 279–93. doi:[10.1007/s00607-005-0139-x](https://doi.org/10.1007/s00607-005-0139-x).*
+*
+* @param N        number of indexed elements
+* @param alpha    scalar constant
+* @param X        input array
+* @param strideX  index increment
+* @param offsetX  starting index
+*/
+double stdlib_strided_dapxsumkbn2_ndarray( const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX  ) {
+	double sum;
+	double ccs;
+	CBLAS_INT ix;
+	CBLAS_INT i;
+	double cs;
+	double cc;
+	double v;
+	double t;
+	double c;
+
+	if ( N <= 0 ) {
+		return 0.0;
+	}
+	if ( N == 1 || strideX == 0 ) {
+		return alpha + X[ 0 ];
+	}
+	ix = offsetX;
+	sum = 0.0;
+	ccs = 0.0; // second order correction term for lost low order bits
+	cs = 0.0; // first order correction term for lost low order bits
+	for ( i = 0; i < N; i++ ) {
+		v = alpha + X[ ix ];
+		t = sum + v;
+		if ( fabs( sum ) >= fabs( v ) ) {
```

Comment:
@headlessNode Would you mind refactoring to use `@stdlib/math/base/special/abs`, rather than `fabs` here? You'll need to add `@stdlib/math/base/special/abs` to the manifest.json file as a dep, replace the `math.h` include above, and update the `fabs` calls here and below.

Reply:
Sure